### DISCUSSION TITLE
Logs are not generating each time for Lwc and Aura

### DISCUSSION THREAD

**Discussion by monika-gupta-erudite** [jongpie/NebulaLogger#581] [Q&A] [answered]
I am exploring this logger for LWC and Aura 
Created a simple LWC to create logs on a button click
![Lwc](https://github.com/jongpie/NebulaLogger/assets/140600429/fc9969b5-897e-4d50-9c59-356a8c911de6)
Logs are not Generating on single click it gets generated on double click of the button.
and generated logEntry record does not have information about the LWC component  
![image](https://github.com/jongpie/NebulaLogger/assets/140600429/457dc92d-c601-4626-adfd-cb49b8363326)
Same thing happening with my Aura component .
LogEntry record generated from Aura component saying Component type as LWC.  
I tried changing the save methods also.

**Comment by jongpie** [comment]
Hi @monika-gupta-erudite - do you know which version of the unlocked package you're using? There was a recent bugfix, [v4.11.12](https://github.com/jongpie/NebulaLogger/releases/tag/v4.11.12), to try to address these issues, but if you're already on that version, there may still be some work needed to fully fix this bug.

**Reply by monika-gupta-erudite** [reply]
Hi @jongpie I am using v4.11.12 only 
![image](https://github.com/jongpie/NebulaLogger/assets/140600429/97799a5b-1b2c-4282-8c41-5dc4071af491)

If I change my code to below code. It solves the issue in both LWC and Aura Component. 
![image](https://github.com/jongpie/NebulaLogger/assets/140600429/4350add5-c628-49ee-a2d9-c99d7456ac3a)

I have one more question
Can I use Logger in the doInit Method of Aura Component?
![image](https://github.com/jongpie/NebulaLogger/assets/140600429/8a181717-ea0b-407a-87f6-64f1ae9125d6)
These logs records are not generating even on the browser console it is not visible.

**Reply by jongpie** [reply]
@monika-gupta-erudite I've created issue #615 to track progress on these issues. The issue related to the component name & function being blank should be fixed now - [see my comment here](https://github.com/jongpie/NebulaLogger/issues/615#issuecomment-1913710474).

**Reply by jongpie** [reply, marked as answer]
One more update here for visibility - these issues should now be fixed in [release `v4.13.0`](https://github.com/jongpie/NebulaLogger/releases/tag/v4.13.0)!
